test/functional/p2p_addrfetch.py

```diff
@@ -45,17 +51,28 @@ def run_test(self):
         msg = msg_addr()
         msg.addrs = [ADDR]
         peer.send_and_ping(msg)
-        assert_equal(len(node.getpeerinfo()), 1)
+        self.assert_getpeerinfo(peer_ids=[peer_id])
 
         self.log.info("Check that answering with larger addr messages leads to disconnect")
         msg.addrs = [ADDR] * 2
         peer.send_message(msg)
         peer.wait_for_disconnect(timeout=5)
 
         self.log.info("Check timeout for addr-fetch peer that does not send addrs")
-        peer = node.add_outbound_p2p_connection(P2PInterface(), p2p_idx=1, connection_type="addr-fetch")
-        node.setmocktime(int(time.time()) + 301)  # Timeout: 5 minutes
+        peer_id = 1
+        peer = node.add_outbound_p2p_connection(P2PInterface(), p2p_idx=peer_id, connection_type="addr-fetch")
+
+        time_now = int(time.time())
+        self.assert_getpeerinfo(peer_ids=[peer_id])
+
+        # Expect addr-fetch peer connection to be maintained up to 5 minutes.
+        node.setmocktime(time_now + 295)
+        self.assert_getpeerinfo(peer_ids=[peer_id])
+
+        # Expect addr-fetch peer connection to be disconnected after 5 minutes.
+        node.setmocktime(time_now + 301)
         peer.wait_for_disconnect(timeout=5)
+        self.assert_getpeerinfo(peer_ids=[])
```

Comment:
I don't think this line is adding much value since `wait_for_disconnect` would fail if the peer was still connected, but if you want to keep it you could consider updating to:
```suggestion
        assert_equal(len(node.getpeerinfo()), 0)
```

since that's the only check that would run with an empty array for `peer_ids`

Reply:
The idea was for it to work with only a change to the value of the `peer_ids` argument if the number of peers changes.

Reply:
(Thanks for having a look!)

Reply:
same as above (https://github.com/bitcoin/bitcoin/pull/22568#discussion_r690508132), I'd prefer if this was the simplest version that captured current expectations.

Reply:
For the essentially the same amount of code, this suggestion could increase the cost of future change (and is also less symmetric with the other changes) so it doesn't seem worth updating for. Prefer to leave as-is.